org.eclipse.jdt.debug.ui/ui/org/eclipse/jdt/internal/debug/ui/actions/RemoveAllCondtionalBreakpoints.java

```diff
@@ -0,0 +1,115 @@
+/*******************************************************************************
+ * Copyright (c) 2024 IBM Corporation.
+ *
+ * This program and the accompanying materials
+ * are made available under the terms of the Eclipse Public License 2.0
+ * which accompanies this distribution, and is available at
+ * https://www.eclipse.org/legal/epl-2.0/
+ *
+ * SPDX-License-Identifier: EPL-2.0
+ *
+ * Contributors:
+ * IBM - Initial API and implementation
+ *******************************************************************************/
+package org.eclipse.jdt.internal.debug.ui.actions;
+
+import org.eclipse.core.resources.IWorkspace;
+import org.eclipse.core.resources.IWorkspaceRunnable;
+import org.eclipse.core.resources.ResourcesPlugin;
+import org.eclipse.core.runtime.CoreException;
+import org.eclipse.core.runtime.IProgressMonitor;
+import org.eclipse.core.runtime.IStatus;
+import org.eclipse.core.runtime.Status;
+import org.eclipse.core.runtime.jobs.Job;
+import org.eclipse.debug.core.DebugPlugin;
+import org.eclipse.debug.core.model.IBreakpoint;
+import org.eclipse.debug.internal.ui.DebugUIPlugin;
+import org.eclipse.debug.internal.ui.actions.breakpoints.RemoveAllTriggerPointsAction;
+import org.eclipse.jdt.debug.ui.IJavaDebugUIConstants;
+import org.eclipse.jdt.internal.debug.core.breakpoints.JavaLineBreakpoint;
+import org.eclipse.jdt.internal.debug.ui.JDIDebugUIPlugin;
+import org.eclipse.jface.action.IAction;
+import org.eclipse.jface.dialogs.IDialogConstants;
+import org.eclipse.jface.dialogs.MessageDialogWithToggle;
+import org.eclipse.jface.preference.IPreferenceStore;
+import org.eclipse.ui.IWorkbenchWindow;
+
+
+public class RemoveAllCondtionalBreakpoints extends RemoveAllTriggerPointsAction {
+	public RemoveAllCondtionalBreakpoints() {
+		super();
+	}
+	@Override
+	protected boolean isEnabled() {
+		for (IBreakpoint breakpoint : DebugPlugin.getDefault().getBreakpointManager().getBreakpoints()) {
+			if (breakpoint instanceof JavaLineBreakpoint javaBreakpoint) {
+				try {
+					if (javaBreakpoint.isConditionEnabled()) {
+						return true;
+					}
+				} catch (CoreException e) {
+					DebugUIPlugin.log(e);
+				}
+			}
+		}
+		return false;
+	}
+
+	@Override
+	public void run(IAction action) {
+		IWorkbenchWindow window = JDIDebugUIPlugin.getActiveWorkbenchWindow();
+		if (window == null) {
+			return;
+		}
+		IPreferenceStore store = JDIDebugUIPlugin.getDefault().getPreferenceStore();
+		boolean prompt = store.getBoolean(IJavaDebugUIConstants.PREF_PROMPT_REMOVE_ALL_CONDITIONAL_BREAKPOINTS);
```

Comment:
Not sure why one would prompt anyone on **disabling** breakpoints. This only makes sense on **removing** breakpoints, which we don't plan to do here.

Reply:
Remove all trigger have the same check, so only tried the same.